crates/wash-cli/src/call.rs

```diff
@@ -222,27 +234,135 @@ pub async fn handle_call(
+                &wrpc_client,
+                &component_id,
+                &lattice,
+                &instance,
+                &name,
+                opts.timeout_ms,
+            )
+            .await
+        }
+    }
+}
 
+/// Utility type used mostly for printing HTTP responses to the console as JSON
+#[derive(Debug, Clone, Serialize)]
+struct HttpResponse {
+    status: u16,
+    headers: HashMap<String, String>,
+    body: Bytes,
+}
+
+/// Invoke a wRPC endpoint that takes a HTTP request (usually `wasi:http/incoming-handler.handle`);
+async fn wrpc_invoke_http_handler(
+    wrpc_client: &wasmcloud_core::wrpc::Client,
+    lattice: &str,
+    component_id: &str,
+    timeout_ms: u64,
+) -> Result<CommandOutput> {
+    // TODO: make this customizable/readable from user CLI input using the CallCommand::input
+    let request = http::Request::builder()
+        .uri("http://localhost:8080")
+        .method("GET")
+        .body(String::new())
+        .context("failed to build HTTP request")?;
+    debug!("successfully built HTTP request");
+
+    let result = tokio::time::timeout(
+        std::time::Duration::from_millis(timeout_ms),
+        wrpc_client.invoke_handle_http(request),
+    )
+    .await
+    .with_context(|| format!("component invocation timeout, is component [{component_id}] running in lattice [{lattice}]?"))?
+    .context("failed to perform HTTP request")?;
+
+    match result {
+        (Ok(mut resp), tx, _body_err) => {
+            tx.await
+                .context("failed to wait for transmission to close")?;
+
+            let status = resp.status().as_u16();
+            let headers =
+                HashMap::<String, String>::from_iter(resp.headers().into_iter().map(|(k, v)| {
+                    (
+                        k.as_str().into(),
+                        v.to_str().map(|v| v.to_string()).unwrap_or_default(),
+                    )
+                }));
+
+            // Read the incoming body into a string
+            let mut body = BytesMut::new();
+            while let Some(Ok(bytes)) = resp.body_mut().body.next().await {
+                body.extend(bytes);
+            }
+            let body = body.freeze();
+
+            let http_resp = HttpResponse {
+                status,
+                headers,
+                body,
+            };
+
+            Ok(CommandOutput::new(
+                serde_json::to_string_pretty(&http_resp)
+                    .context("failed to print http response JSON")?,
+                HashMap::from([(
+                    "response".into(),
+                    serde_json::to_value(&http_resp)
+                        .context("failed to convert http response to value")?,
+                )]),
+            ))
+        }
+        // All other responses we're not
+        _ => bail!("unexpected response after HTTP wRPC invocation"),
+    }
+}
+
+/// Invoke a wRPC endpoint that takes nothing and returns a string
+async fn wrpc_invoke_string_echo(
```

Comment:
nit: `echo` feels a little misleading here since it's calling a no-arg function but I don't have a better suggestion than like `wrpc_invoke_simple`

Reply:
Yup, renamed!